### ISSUE TITLE
`never_loop` false positive with conditional `continue`

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang-nursery/rust-clippy#2088]
This code snippet prints "Looping" forever ([full crate](https://github.com/Aaron1011/clippy_bug)):

```rust
fn main() {
    let mut a = true;
    loop {
        while a {
            if true {
                a = false;
                continue;
            }
            return;
        }
        println!("Looping");
    }
}
```

However, the `never_loop` lint incorrectly marks the outer loop as never looping:

```
warning: this loop never actually loops
  --> src/main.rs:3:5
   |
3  | /     loop {
4  | |         while a {
5  | |             if true {
6  | |                 a = false;
...  |
11 | |         println!("Looping");
12 | |     }
   | |_____^
   |
   = note: #[warn(never_loop)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.164/index.html#never_loop
```

**Comment by oli-obk** [comment]
> Can I just compare two NodeId's with a '<' and expect the outer loop to have a lower value?

No, `NodeId`s are random, do not expect them to have any semantic meaning

> The problem is that this is actually only true if the continue/break's destination is outside (not inside) of the loop in question. 

I'm not sure I understand, can you give code examples?

**Comment by oli-obk** [comment]
Thanks for the explanation, that makes total sense
